### PR TITLE
Frontend setup

### LINKED ISSUE #50: Add initial react frontend
- Table of random numbers
- Button for creating a random number and refreshing the table

### PR DESCRIPTION
**Pull request by kujiraOo**
Closes #50 #95

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -17,8 +17,32 @@ jobs:
...
       run: npm run lint
 
-  test:
+  test-backend:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - run: docker-compose -f docker-compose.test.yml run test-ci
+
+  lint-frontend:
```
[the remark this answers is not in the document]
Comment (kujiraOo) [reply .github/workflows/main.yml:26]: Didn't find any simple way to do it

File: frontend/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "frontend",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.12.0",
```
[the remark this answers is not in the document]
Comment (kujiraOo) [reply frontend/package.json]: Done

File: frontend/src/index.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  <React.StrictMode>
+    <App />
+  </React.StrictMode>,
+  document.getElementById('root'),
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
[the remark this answers is not in the document]
Comment (kujiraOo) [reply frontend/src/index.jsx]: `reportWebVitals` is used for collecting various performance metrics, we don't need it for now

File: frontend/src/utils/client.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      'Content-Type': 'application/json',
+    },
+    ...(body && { body: JSON.stringify(body) }),
+  });
+
+  if (!response.ok) {
+    const { status, statusText } = response;
+
+    // eslint-disable-next-line no-console
+    console.error(`request failed ${status} ${statusText}`);
```
[the remark this answers is not in the document]
Comment (kujiraOo) [reply frontend/src/utils/client.js:14]: Actually, let's keep it as it is for now, we can implement some centralized error handling later

**Review by PenguinSatan — APPROVED** [review]

**Pull request merged**